lib/Doctrine/ODM/PHPCR/Mapping/ClassMetadata.php

```diff
@@ -329,6 +329,13 @@ class ClassMetadata implements ClassMetadataInterface
     public $parentClasses = array();
 
     /**
+     * READ-ONLY: Child class restrictions.
+     *
+     * @var array
+     */
+    public $childRestrictions = null;
```

Comment:
if we were to follow from PHPCR we could just have `requiredClasses` as with `NodeDefinition::getRequiredPrimaryTypes`